service/history/mutableStateBuilder.go

```diff
@@ -513,6 +512,33 @@ func (e *mutableStateBuilder) CreateReplicationTask(newRunEventStoreVersion int3
 	return t
 }
 
+func (e *mutableStateBuilder) checkAndClearTimerFiredEvent(timerID string) *workflow.HistoryEvent {
+	var timerEvent *workflow.HistoryEvent
+	e.bufferedEvents, timerEvent = checkAndClearTimerFiredEvent(e.bufferedEvents, timerID)
+	if timerEvent != nil {
+		return timerEvent
+	}
```

Comment:
not sure whether also need to check updateBufferedEvents

Reply:
I was not sure about this. I think it doesnt harm to also check updateBufferedEvents here. I can add a check for that too..